core/node/api_server/src/execution_sandbox/execute.rs

```diff
@@ -171,7 +204,53 @@ impl SandboxExecutor {
             options,
             storage_caches: Some(caches),
             timestamp_asserter_params,
+            vm_divergence_counter,
+        }
+    }
+
+    pub(crate) fn vm_mode(&self) -> FastVmMode {
+        self.options.fast_vm_mode
+    }
+
+    pub(crate) fn vm_divergence_counter(&self) -> Arc<AtomicUsize> {
+        self.vm_divergence_counter.clone()
+    }
+
+    async fn dump_vm_state(
+        last_dump_timestamp: &Mutex<Option<Instant>>,
+        store: &dyn ObjectStore,
+        vm_dump: VmDump,
+    ) -> anyhow::Result<()> {
+        /// Minimum interval between VM dumps.
+        const DUMP_INTERVAL: Duration = Duration::from_secs(10);
```

Comment:
Is this policy (saving no more than one VM dump per 10s OK? Maybe it makes sense to make the interval configurable or be able to disable saving VM dumps altogether? Also, do our API server pods in K8s have the necessary GCS authentication data? (If this is a security concern, it could be possible to use a local filesystem object store for VM dumps, but this has obvious downsides, like dumps disappearing after a pod restart and being inaccessible by devs for the Era mainnet etc.)

Reply:
Most likely APIs don't have access to GCS currently. IMO, 10s is too frequent, 1-10 minutes is more reasonable. I wouldn't make it configurable to keep it simpler. I'm worried about number of dumps growing in the storage, e.g. if we introduce a regression then we will have a lot dumps, probably we some info alert or another process to monitor it